src/cluster_legacy.c

```diff
@@ -5680,7 +5695,7 @@ void addShardReplyForClusterShards(client *c, list *nodes) {
     addReplyBulkCString(c, "slots");
 
     /* Use slot_info_pairs from the primary only */
-    n = clusterNodeGetPrimary(n);
+    n = clusterNodeGetHealthyPrimary(n);
```

Comment:
something like this and I don't think we need a new function
```suggestion
    listIter li;
    listNode *ln;
    list *nodes = clusterGetNodesInMyShard(node);
    listRewind(nodes, &li);
    while ((ln = listNext(&li))) {
        clusterNode *node = listNodeValue(ln);
        if (node->slot_info_pairs) {
            break;
        }
    }
```

Reply:
Yeah, we could do this.